pkg/handlers/embedded_cluster_node_join_command.go

```diff
@@ -178,13 +169,37 @@ func (h *Handler) GetEmbeddedClusterNodeJoinCommand(w http.ResponseWriter, r *ht
 		return
 	}
 
-	JSON(w, http.StatusOK, GetEmbeddedClusterNodeJoinCommandResponse{
-		ClusterID:              install.Spec.ClusterID,
+	clusterUUID, err := uuid.Parse(install.Spec.ClusterID)
+	if err != nil {
+		logger.Error(fmt.Errorf("failed to parse cluster id: %w", err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	var currentAppVersionLabel string
+	// attempt to get the current app version label from the installed app
+	installedApps, err := store.GetStore().ListInstalledApps()
+	if err == nil && len(installedApps) > 0 {
+		appVersion, err := store.GetStore().GetAppVersion(installedApps[0].ID, installedApps[0].CurrentSequence)
```

Comment:
i know it's a weird name, but `CurrentSequence` here doesn't mean currently deployed, it means the latest available version for the app in the admin console / version history page.

Reply:
oh well that would cause a problem
thanks for catching this

Reply:
in my testing the version returned is the deployed version, not the latest available version

Reply:
https://github.com/replicatedhq/kots/blob/0589dd2ff354b3d2c0a688663abafc54b397476f/pkg/store/kotsstore/downstream_store.go#L142

I think the code bears this out too

Reply:
that's not where app.CurrentSequence comes from. it comes from the app table. in your testing, did you have a newer non-deployed version that has a different version label?

https://github.com/replicatedhq/kots/blob/5727dfdae28d6011a0c94f368caed884f75bcaac/pkg/store/kotsstore/app_store.go#L207

Reply:
Yes, I have a newer version available: 
<img width="820" alt="Screenshot 2025-04-16 at 11 51 56" src="https://github.com/user-attachments/assets/98753059-3a1f-4ba3-90cc-cc1bdb552e5f" />

But the install response includes "appver-dev-36Xpeq" not "local-test"

Reply:
and rollbacks aren't supported in EC, so the app current sequence will always be the deployed sequence. i'm fine with that, just need to add a comment explaining this... not sure if we plan to support rollbacks in the future